gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/DefaultPubSubRpc.java

```diff
@@ -50,48 +54,78 @@
 import com.google.pubsub.v1.Subscription;
 import com.google.pubsub.v1.Topic;
 
+import io.grpc.ManagedChannel;
+import io.grpc.Status.Code;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+
 import org.joda.time.Duration;
 
 import java.io.IOException;
 import java.util.Set;
 import java.util.concurrent.Future;
-
-import autovalue.shaded.com.google.common.common.collect.Sets;
-import io.grpc.Status.Code;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
 
 public class DefaultPubSubRpc implements PubSubRpc {
 
   private final PublisherApi publisherApi;
   private final SubscriberApi subscriberApi;
+  private final ScheduledExecutorService executor =
+      MoreExecutors.getExitingScheduledExecutorService(new ScheduledThreadPoolExecutor(8));
 
   public DefaultPubSubRpc(PubSubOptions options) throws IOException {
     try {
-      // Provide (and use a common thread-pool).
-      // This depends on https://github.com/googleapis/gax-java/issues/73
-      PublisherSettings.Builder pbuilder =
-          PublisherSettings.defaultBuilder()
-              .provideChannelWith(options.authCredentials().credentials())
-              .applyToAllApiMethods(apiCallSettings(options));
-      publisherApi = PublisherApi.create(pbuilder.build());
-      SubscriberSettings.Builder sBuilder =
-          SubscriberSettings.defaultBuilder()
-              .provideChannelWith(options.authCredentials().credentials())
-              .applyToAllApiMethods(apiCallSettings(options));
-      subscriberApi = SubscriberApi.create(sBuilder.build());
+      PublisherSettings.Builder pubBuilder =
+          PublisherSettings.defaultBuilder().provideExecutorWith(executor, false);
+      SubscriberSettings.Builder subBuilder =
+          SubscriberSettings.defaultBuilder().provideExecutorWith(executor, false);
+      // todo(mziccard): PublisherSettings should support null/absent credentials for testing
```

Comment:
Most users will have `gcloud beta auth application-default login` of some kind set, and all of our testing environments should.  So this might not be necessary.